### DISCUSSION TITLE
How would I add CSS classes to Pandoc `Block` nodes?

### DISCUSSION THREAD

**Discussion by solomon-b** [jgm/pandoc#10500] [Q&A]
Im using Pandoc as a library in the backend of a website where I want to take markdown documents and render them as HTML. I"m using TailwindCSS for styling which means that I need to be able to add a custom CSS classes to the generated HTML elements. 

Some of the `Block` constructors contain a `Attr` parameter which allows me to insert a `class` attribute via a recursive function. However, this is not the case for all `Block` constructors. Why is that? Is there some other way to insert attributes, either into the Pandoc AST or via the HTML writer?

**Comment by massifrg** [comment]
You can use `Div` elements around the blocks without `Attr` that you want to customize.

`Div` elements with a `custom-style` attribute are used, for example, to add custom styles to the paragraphs they enclose (only in some output formats, like DOCX, ICML, ODT).

```markdown
::: {custom-style="myStyle"}
A paragraph.

Another paragraph.
:::
```

In this example, you get two paragraphs with "myStyle" custom style.

You can do something similar, but you must write a filter that uses that information to add classes to the HTML elements inside those `Div`s, when you convert from markdown to HTML.
